test/cpp/api/functional.cpp

```diff
@@ -360,3 +360,24 @@ TEST_F(FunctionalTest, LogSigmoid) {
   auto y_exp = torch::log(torch::ones_like(x)/(torch::ones_like(x) + torch::exp(torch::neg(x))));
   ASSERT_TRUE(torch::allclose(y, y_exp, 1e-4, 1e-7));
 }
+
+TEST_F(FunctionalTest, Normalize) {
+  {
+    auto input = torch::randn({1,3,4,4}, torch::requires_grad());
+    auto norm = F::normalize(input, NormalizeOptions().p(1).dim(-1));
+    // reduce to scalar to call .backward()
+    torch::Tensor s = norm.sum();
+    s.backward();
+    ASSERT_EQ(s.ndimension(), 0);
```

Comment:
It would be great to assert that `input.grad().numel()` is as expected, to show that the backward pass is successful.

Reply:
thanks this was the missing key and was confusing me quite a bit because input.grad() is none if you apply a reshape operation and override the identifier that holds the original unreshaped variable. arange and linspace don't have ways to specify size into their factory functions, I'm still now sure how this works in C++

My understanding as of now 

```python
input = torch.arange(10, dtype=float, requires_grad=True).reshape((1,2,5))
loss = input.norm().sum()
loss.backward()

input.grad
=> None
```

I suspect this happens because even though autograd is tracking the original variable the identifier to it lost in the reshape operation which does not have a derivative (I believe?)

Reply:
I think your assessment is correct, and we can use `input = torch::tensor({{{0, 1, 2, 3, 4}, {5, 6, 7, 8, 9}}}, torch::dtype(torch::kFloat).requires_grad(true))` as tensor to get around this issue. :D

Reply:
That is much neater I agree xD